CRM/Sqltasks/Task.php

```diff
@@ -425,9 +469,30 @@ public function execute($params = []) {
       $this->status = 'success';
     }
 
+    $this->logExecutionTask($task_runtime, $taskStartDate, $inputValue);
+
     return $this->log_messages;
   }
 
+  /**
+   * Logs the data after while execution
+   *
+   * @return void
+   */
+  public function logExecutionTask($taskRuntime, $taskStartDate, $inputValue) {
+    CRM_Sqltasks_BAO_SqltasksExecution::create([
```

Comment:
We don't need to change it here now that it's already working, but generally speaking I would prefer if we interact with the DB primarily through API4.